corehq/apps/users/models.py

```diff
@@ -230,6 +230,11 @@ class UserRole(Document):
     permissions = SchemaProperty(Permissions)
     is_archived = BooleanProperty(default=False)
 
+    @classmethod
+    @quickcache(['doc_id'], timeout=0)
```

Comment:
@czue not sure if you're thinking of `@memoized` which is optimized to live only as long as the `self` instance on an instance method, but `@quickcache` (I'm pretty sure) doesn't distinguish between functions, methods, class methods, static methods—even in memory it's all just key-based, with the key defined by your `vary_on` variables.

Still the point stands that this will live longer than the current request cycle, regardless of whether you're doing it in memory or in redis (or both).